sklearn/decomposition/sparse_pca.py

```diff
@@ -116,6 +136,17 @@ def fit(self, X, y=None):
         """
         random_state = check_random_state(self.random_state)
         X = check_array(X)
+
+        if self.normalize_components:
+            self.mean_ = np.mean(X, axis=0)
```

Comment:
I would almost call `X.mean(axis=0)` but it is true that we don't support sparse matrix